src/test/java/core/basesyntax/service/ParserServiceImplTest.java

```diff
@@ -0,0 +1,70 @@
+package core.basesyntax.service;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.Transaction;
+import core.basesyntax.service.impl.ParserServiceImpl;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ParserServiceImplTest {
+    private static final String OPERATION_TYPE = "r";
+    private static final String FRUIT = "banana";
+    private static final Integer QUANTITY = 20;
+    private static final String VALID_INPUT = "r,banana,20";
+    private static final String INVALID_INPUT_DIGIT = "r,banana,b";
+    private static final String INVALID_INPUT_PUNCTUATION = "r.banana.b";
+    private static final String INVALID_INPUT_WRONG_ORDER = "20,r.banana";
+    private static final String HEADER = "type,fruit,quantity";
```

Comment:
Hardcoding is allowed in the tests. So my advice is to write it all at once if you only use it once.